network/oauth_client.go

```diff
@@ -64,6 +64,7 @@ func (oc OAuthClient) Do(request *http.Request) (*http.Response, error) {
 	}
 
 	client := oc.oauthConfig.Client(oc.context, token)
+	client.Timeout =  oc.timeout
```

Comment:
It won't be respected if it is added in http.Client, as explained in my comment for this PR. This is because in oauth2.NewClient it only inherits the Transport section of the Client thats passed in.